components/CollectivePickerAsync.js

```diff
@@ -69,7 +73,7 @@ const Messages = defineMessages({
  */
 const getPlaceholder = (intl, types) => {
   const nbTypes = types ? types.length : 0;
-  if (nbTypes === 0 || nbTypes > 3) {
+  if (nbTypes === 0 || nbTypes > 4) {
```

Comment:
Limiting to 3 was done on purpose to prevent responsiveness issues. Are you sure this won't cause any regression?

Reply:
I don't see the justification for the change in this PR.

Reply:
The reason I did this change is to allow `Funds` to appear int the placeholder as well. I don't think it creates a regression; I've tested on small screen sizes as well and it just appears fine; 

![image](https://user-images.githubusercontent.com/12435965/155346877-572d2672-6c70-4cb2-8dfb-a174374c7a28.png)

Reply:
> I don't see the justification for the change in this PR.

@znarf : Meaning we don't want the placeholder update, even-though funds appear in the collective picker?

Reply:
If there is a justification, I'm ok with it if there is no regression.

Reply:
This doesn't look good to me, it's crowded in there. Again, limiting to 3 was a conscious decision and I think we should stick with it, it doesn't hurt to keep just "Search" when there are more types. We can discuss with design if we want to review this, but it's out of scope for this PR.

![image](https://user-images.githubusercontent.com/1556356/155510902-c61c343b-e917-44ab-a4b9-dd76c29d297b.png)

Reply:
I've reverted that change so that we now show just `Search` when the list is longer than 3. I should mention though that this also has the crowded issue on some mobiles (although very few); such as the Galaxy S20;

![image](https://user-images.githubusercontent.com/12435965/155547451-57abc991-b4c1-4c1f-b637-fac639bc87a7.png)